pxtservices/iframeEmbeddedClient.ts

```diff
@@ -0,0 +1,79 @@
+type IframeClientSetMessagePortRequest = {
+    type: "iframeclientsetmessageport";
+};
+
+type IFrameClientReadyMessage = {
+    type: "iframeclientready";
+};
+
+export type IframeClientMessage = IframeClientSetMessagePortRequest | IFrameClientReadyMessage;
+
+export class IFrameEmbeddedClient {
+    protected frameId: string;
+    protected port: MessagePort;
+
+    constructor(protected messageHandler: (message: MessageEvent) => void) {
+        this.frameId = frameId();
+
+        window.addEventListener("message", this.onMessageReceived);
```

Comment:
Provide a mechanism to remove this listener? If the containing object is unmounted, for instance, this service should be able to be stopped.